src/MongoDB.Driver.Core/Core/Authentication/ScramSha1Authenticator.cs

```diff
@@ -55,7 +55,7 @@ public ScramSha1Authenticator(UsernamePasswordCredential credential)
         }
 
         internal ScramSha1Authenticator(UsernamePasswordCredential credential, IRandomStringGenerator randomStringGenerator)
-            : base(credential, HashAlgorithmName.SHA1, randomStringGenerator, H1, Hi1, Hmac1) 
+            : base(credential, HashAlgorithmName.SHA1, randomStringGenerator, H1, Hi1, Hmac1, new ScramCache()) 
```

Comment:
I'm a little confused by where the cache is created and how it's past around.

It's probably correct, but merits careful attention.

Reply:
In this specific example, I added the cache to the base ScramShaAuthenticator class, so when ScramSha1 and ScramSha256 are constructed, they create new caches to pass into the base constructor. Those become the caches that we use later on.

More generally, the idea was to put the cache in the authenticator, like we discussed offline. The actual computations we want to use the cache for happen in the internal ClientFirst class, which makes it a bit tricky to pass the cache down and make sure it's being used properly.